src/plugins/log_files/include/plugins/log_files/log_files.h

```diff
@@ -153,12 +153,28 @@ class LogFiles : public PluginBase {
      * @brief Callback type for download_log_file_async.
      */
 
-    using DownloadLogFileCallback = std::function<void(Result, ProgressData)>;
+    using DownloadLogFileCallback = std::function<void(LogFiles::Result, ProgressData)>;
 
     /**
      * @brief Download log file.
+     *
+     * This function is non-blocking. See 'download_log_file' for the non-blocking counterpart.
+     */
+    void download_log_file_async(LogFiles::Entry entry, std::string path, DownloadLogFileCallback callback);
+
+    /** @brief Download log file.
+     *
+     * This function is blocking. See 'download_log_file_async' for the non-blocking counterpart.
+     *
+     * @return Result of the log download.
+     */
+
+    LogFiles::Result download_log_file(LogFiles::Entry entry, std::string path);
+
+    /**
+     * @brief Erase all log files.
      */
-    void download_log_file_async(Entry entry, std::string path, DownloadLogFileCallback callback);
+    void erase_log_files();
```

Comment:
Note that the plugin.h and plugin.cpp files are auto-generated from proto.
If you did in fact auto-generate it from proto, then you just need to make a PR with the proto additions as well here: https://github.com/mavlink/MAVSDK-Proto and then update the submodule in this repo.

See: https://mavsdk.mavlink.io/main/en/cpp/contributing/plugins.html#add-api-to-proto

Reply:
In fact I did not. Thanks for the links; I've added a PR to the protos repo.